engines/wage/world.cpp

```diff
@@ -0,0 +1,510 @@
+		_commandsMenuName = menu->operator[](0);
+		_commandsMenu = menu->operator[](1);
+		delete menu;
+		delete res;
+	}
+	res = resMan->getResource(MKTAG('M','E','N','U'), 2005);
+	if (res != NULL) {
+		Common::StringArray *menu = readMenu(res);
+		_weaponsMenuName = menu->operator[](0);
+		delete menu;
+		delete res;
+	}
+	// TODO: Read Apple menu and get the name of that menu item..
+
+	// store global info in state object for use with save/load actions
+	//world.setCurrentState(initialState);	// pass off the state object to the world
+
+	return true;
+}
+
+Common::StringArray *World::readMenu(Common::SeekableReadStream *res) {
+	res->skip(10);
+	int enableFlags = res->readUint32BE();
+	Common::String menuName = readPascalString(res);
+	Common::String menuItem = readPascalString(res);
+	int menuItemNumber = 1;
+	Common::String menu;
+	byte itemData[4];
+
+	while (menuItem.size() > 0) {
+		if (menu.size() > 0) {
+			menu += ';';
+		}
+		if ((enableFlags & (1 << menuItemNumber)) == 0) {
+			menu += '(';
+		}
+		menu += menuItem;
+		res->read(itemData, 4);
+		static const char styles[] = {'B', 'I', 'U', 'O', 'S', 'C', 'E', 0};
+		for (int i = 0; styles[i] != 0; i++) {
+			if ((itemData[3] & (1 << i)) != 0) {
+				menu += '<';
+				menu += styles[i];
+			}
+		}
+		if (itemData[1] != 0) {
+			menu += '/';
+			menu += (char)itemData[1];
+		}
+		menuItem = readPascalString(res);
+		menuItemNumber++;
+	}
+
+	Common::StringArray *result = new Common::StringArray;
+	result->push_back(menuName);
+	result->push_back(menu);
+
+	debug(4, "menuName: %s", menuName.c_str());
+	debug(4, "menu: %s", menu.c_str());
+
+	return result;
+}
+
+void World::loadExternalSounds(Common::String fname) {
+	Common::File in;
+
+	in.open(fname);
+	if (!in.isOpen()) {
+		warning("Cannot load sound file <%s>", fname.c_str());
+		return;
+	}
+	in.close();
+
+	Common::MacResManager *resMan;
+	resMan = new Common::MacResManager();
+	resMan->open(fname);
+
+	Common::MacResIDArray resArray;
+	Common::SeekableReadStream *res;
+	Common::MacResIDArray::const_iterator iter;
+
+	resArray = resMan->getResIDArray(MKTAG('A','S','N','D'));
+	for (iter = resArray.begin(); iter != resArray.end(); ++iter) {
+		res = resMan->getResource(MKTAG('A','S','N','D'), *iter);
+		addSound(new Sound(resMan->getResName(MKTAG('A','S','N','D'), *iter), res));
+	}
+}
+
+Common::String *World::loadStringFromDITL(Common::MacResManager *resMan, int resourceId, int itemIndex) {
+	Common::SeekableReadStream *res = resMan->getResource(MKTAG('D','I','T','L'), resourceId);
+	if (res) {
+		int itemCount = res->readSint16BE();
+		for (int i = 0; i <= itemCount; i++) {
+			// int placeholder; short rect[4]; byte flags; pstring str;
+			res->skip(13);
+			Common::String message = readPascalString(res);
+			if (i == itemIndex) {
+				Common::String *msg = new Common::String(message);
+				delete res;
+				return msg;
+			}
+		}
+
+		delete res;
+	}
+
+	return NULL;
+}
+
+bool InvComparator(Obj *l, Obj *r) {
```

Comment:
This is a function thus `invComparator` according to our guidelines. It can be made `static` too.